src/app/pages/OriginsLaunchpad/pages/SalesDay/index.tsx

```diff
@@ -8,42 +16,35 @@ import { useIsConnected } from 'app/hooks/useAccount';
 import { ImportantInformationStep } from './pages/ImportantInformationStep';
 import { BuyStep } from './pages/BuyStep';
 import { useGetSaleInformation } from '../../hooks/useGetSaleInformation';
+import saleStorage from './storage';
 import styles from './index.module.scss';
 
 interface ISalesDayProps {
   saleName: string;
 }
 
 export const SalesDay: React.FC<ISalesDayProps> = ({ saleName }) => {
+  const match = useRouteMatch();
+  const history = useHistory();
+  const location = useLocation();
   const { t } = useTranslation();
   const connected = useIsConnected();
   const info = useGetSaleInformation();
 
-  const [step, setStep] = useState(1);
+  const setStep = (step: number) => {
+    saleStorage.saveData({ step });
+    history.push(`${match.url}/${step}`);
+  };
 
-  const getActiveStep = (step: number) => {
-    switch (step) {
-      case 1:
-        return (
-          <AccessCodeVerificationStep
-            saleName={saleName}
-            onVerified={() => setStep(2)}
-            info={info}
-          />
-        );
-      case 2:
-        return (
-          <ImportantInformationStep
-            saleName={saleName}
-            onSubmit={() => setStep(3)}
-          />
-        );
-      case 3:
-        return <BuyStep saleInformation={info} saleName={saleName} />;
-      default:
-        return <EngageWalletStep saleName={saleName} />;
+  useEffect(() => {
+    const { step } = saleStorage.getData();
+    if (connected) {
+      history.push(`${match.url}/${step}`);
+    } else {
+      history.push(`${match.url}/engage-wallet`);
     }
-  };
+    // eslint-disable-next-line
```

Comment:
Why do you have eslint ignore here?

Reply:
Please make the requested changes to props pulled from `useRouteMatch()` further down this file, then remove this line and add `url` and `history` to dependency array to satisfy compiler.

Reply:
okay I got it.